R/idr0021 - Segmentation.ipynb

```diff
@@ -0,0 +1,702 @@
+    "    # Calculate the features\n",
+    "    shapes = suppressMessages(computeFeatures.shape(threshImg))\n",
+    "    moments = suppressMessages(computeFeatures.moment(threshImg))\n",
+    "    features <- merge(shapes, moments, by=0, all=TRUE)\n",
+    "    \n",
+    "    if (length(features[,1])>1) {\n",
+    "        # Add the ROIs to the image\n",
+    "        rois <- createROIs(features)\n",
+    "        if (saveROIs)\n",
+    "            addROIs(image, coords = rois)\n",
+    "        \n",
+    "        # Add the interesting properties (area, perimeter and diameter)\n",
+    "        # of the largest ROI together with channel name, image name, image id \n",
+    "        # and roi index to the dataframe\n",
+    "        features <- features[order(-features[,2]),]\n",
+    "        diameter <- features[1,4]*2\n",
+    "        df <- rbind(df, c(channelName, image@dataobject$getName(), image@dataobject$getId(), features[1,1], features[1,2], features[1,3], diameter))\n",
+    "    }\n",
+    "    return(df)\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "datasetId <- REPLACE_WITH_DATASET_ID # Insert the ID of your R-dataset here!\n",
+    "\n",
+    "channelName <- 'CDK5RAP2-C'\n",
+    "\n",
+    "dataset <- loadObject(server, \"DatasetData\", datasetId)\n",
+    "\n",
+    "# Keep the channel name, image name, image id, area, perimeter, and diameter of the largest ROIs\n",
+    "result <- data.frame(Channel = c('remove'), ImageName = c('remove'), Image = c(0), ROIIndex = c(0), Area = c(0), Perimeter = c(0), Diameter = c(0), stringsAsFactors = FALSE)\n",
+    "\n",
+    "images <- getImages(dataset)\n",
+    "for (image in images) {\n",
+    "    result <- tryCatch({\n",
+    "        analyzeImage(image, channelName, result, saveROIs = TRUE)\n",
+    "    }, warning = function(war) {\n",
+    "        message(paste(\"WARNING:  \", image@dataobject$getId(),war))\n",
+    "        return(result)\n",
+    "    }, error = function(err) {\n",
+    "        message(paste(\"ERROR:  \", image@dataobject$getId() ,err))\n",
+    "        return(result)\n",
+    "    }, finally = {\n",
+    "    })\n",
+    "}\n",
+    "\n",
+    "result <- result[-1,]\n",
+    "rownames(result) <- c()\n",
+    "\n",
+    "# set the correct datatypes\n",
+    "result$Channel <- as.factor(result$Channel)\n",
+    "result$Area <- as.numeric(result$Area)\n",
+    "result$Perimeter <- as.numeric(result$Perimeter)\n",
+    "result$Diameter <- as.numeric(result$Diameter)\n",
+    "\n",
+    "result"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Exercise:** Check the ROIs in OMERO.web"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Exercise:** The measurements (Area, Perimeter and Diameter) are currently showing pixels. Find out the pixel size in nanometer and apply it accordingly."
```

Comment:
Where should I find the pixel size in nanometers ? In OMERO.web ? Write it please.

``Apply it accordingly...`` -> Do you mean I should then find a place in the code where the calibration is being applied and change it, then rerun the cell ?

Reply:
Discussed with @dominikl : 

1. highlight this Exercies as (advanced)
2. formulate something like ``add into the cell a code snippet which transforms the pixel values to nanometers. Then rerun the cell.``